src/foam/blob/InputStreamBlob.java

```diff
@@ -50,6 +50,33 @@ public long read(OutputStream out, long offset, long length) {
     }
   }
 
+
+  public String readAsString(OutputStream out, long offset, long length) {
+    try {
+      if ( offset != pos_ ) {
+        throw new RuntimeException("Offset does not match stream position");
+      }
+      ByteArrayOutputStream data = new ByteArrayOutputStream();
+
+      long read = 0;
+      byte[] buffer = new byte[BUFFER_SIZE];
```

Comment:
Is there no way to read from a Stream without an artificial hard limit?

Reply:
Kind of. It can be wrapped with another stream reader which probably will use the same approach inside, because all available read methods that I could find are:
1. read() - Reads a single character
2. read(char[] cbuf, int offset, int length) - Reads characters into a portion of an array.